### DISCUSSION TITLE
Conflict: Multiple assets emit different content to the same filename img/chart.png

### DISCUSSION THREAD

**Discussion by Rolanddoda** [quasarframework/quasar#11709] [General - Components / Directives / etc] [answered]
**Describe the bug**
I just copy-pasted an image into a folder named `chart.png` and I am using it like this:

```html
<q-img :src="require('./chart.png')"></q-img>
```

Now in another folder, I use a completely different image but with the same name `chart.png`.

[Codesandbox](https://codesandbox.io/s/crimson-grass-bq6bp?file=/src/App.vue)

**To Reproduce**
1. Click the Codesandbox link
2. See the error
3. See that component-1 and component-2 use a completely different image.
4. The issue comes only because they have the same name

**Expected behavior**
It shouldn't be an issue since the images are completely different and in different folders.

**Platform (please complete the following information):**
Quasar Version: ^2.3.4
@quasar/app Version: ^3.2.3

Quasar mode:
  - [x] SPA
  - [ ] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX
  - [ ] Vite Plugin

Additonal Info:
- [ ] Typescript
- [ ] SFC usage with `<script setup>`
- [ ] Vuex
- [ ] Prettier (eslint)
- [ ] Standard (eslint)
  
Tested on:
  - [x] SPA
  - [ ] SSR
  - [ ] PWA
  - [ ] Electron
  - [ ] Cordova
  - [ ] Capacitor
  - [ ] BEX

OS: Win 11
Node: 16.13.0
NPM: 8.3.0
Yarn:
Browsers:
iOS:
Android:
Electron:

**Additional context**
I know that images usually should be put in `assets` or `public` folder but for me it's not a problem loading them as modules.

**Comment by Rolanddoda** [comment]
@pdanpdan any reason why this is moved to discussion ?

**Comment by pdanpdan** [comment]
It's a help request for something related most probably to webpack. It's not related to quasar

**Reply by Rolanddoda** [reply]
@pdanpdan I don't know how and why you think it's a help request when it's clear that it's an issue.
If it was for help, I could just name the images to a different name and move on.
But I opened the issue because I thought that you should not get an error for a similar name for an image in completely different folders.

I know that the issue is related to webpack but I haven't changed anything related to it.

**Comment by Rolanddoda** [comment]
@pdanpdan you thought it's not an issue without even asking. And actually it's the second time you are a bit rude with me. [reference](https://github.com/quasarframework/quasar/issues/7531).

But it's ok, no problem at all. Just a friendly request if you can be a bit more helpful and collaborative with the community.
I know that it's easy to close an issue and that's it. But the people who open issues, offer a reproduction, spend their time to help Quasar or ask for help using Quasar. In both cases it should be treated with respect and have a little bit of discussion before making a decision to close an issue.

Anyway, I am not trying to be smart or something just wanted to share my opinion.

Speaking of the issue, I opened a pull-request which seems to be fixing it.
https://github.com/quasarframework/quasar/pull/11711

**Comment by Rolanddoda** [comment, marked as answer]
For people who just want a workaround, modify `quasar.config.js`:

```
    build: {
      // ...
      chainWebpack (chain) {
        chain.module.rule('images')
          .use('url-loader')
          .tap((options) => {
            options.name = 'img/[path][name]${fileHash}.[ext]'
            return options
          })
      }
    }
```
